



-------------- 2016-11-03 10:39:10

2016-11-03 10:37:13 [13][error][abc\frontend\components\ErrorHandler::logException] exception 'yii\base\ErrorException' with message 'getimagesize(): Read error!' in /niub/www/sourcecode/wx-dev.yqtz8.com/frontend/components/InviteManager.php:52
Stack trace:
#0 [internal function]: yii\base\ErrorHandler->handleError(8, 'getimagesize():...', '/niub/www/sourc...', 52, Array)
#1 /niub/www/sourcecode/wx-dev.yqtz8.com/frontend/components/InviteManager.php(52): getimagesize('/niub/www/sourc...')
#2 /niub/www/sourcecode/wx-dev.yqtz8.com/frontend/components/InviteManager.php(78): frontend\components\InviteManager::loadImage('/niub/www/sourc...')
#3 /niub/www/sourcecode/wx-dev.yqtz8.com/frontend/components/InviteManager.php(177): frontend\components\InviteManager::getAndSaveImage('http://wx.qlogo...')
#4 /niub/www/sourcecode/wx-dev.yqtz8.com/frontend/components/event/EventBootstrap.php(69): frontend\components\InviteManager::createInviteImage(Object(frontend\models\Mission))
#5 [internal function]: frontend\components\event\EventBootstrap::handleUpdateMissionPoster(Object(frontend\components\event\AppEvent))
#6 /niub/www/sourcecode/wx-dev.yqtz8.com/vendor/yiisoft/yii2/base/Component.php(541): call_user_func(Array, Object(frontend\components\event\AppEvent))
#7 /niub/www/sourcecode/wx-dev.yqtz8.com/frontend/controllers/ajax/MissionController.php(105): yii\base\Component->trigger('update_mission_...', Object(frontend\components\event\AppEvent))
#8 [internal function]: frontend\controllers\ajax\MissionController->actionPoster()
#9 /niub/www/sourcecode/wx-dev.yqtz8.com/vendor/yiisoft/yii2/base/InlineAction.php(55): call_user_func_array(Array, Array)
#10 /niub/www/sourcecode/wx-dev.yqtz8.com/vendor/yiisoft/yii2/base/Controller.php(154): yii\base\InlineAction->runWithParams(Array)
#11 /niub/www/sourcecode/wx-dev.yqtz8.com/vendor/yiisoft/yii2/base/Module.php(454): yii\base\Controller->runAction('poster', Array)
#12 /niub/www/sourcecode/wx-dev.yqtz8.com/vendor/yiisoft/yii2/web/Application.php(87): yii\base\Module->runAction('ajax/mission/po...', Array)
#13 /niub/www/sourcecode/wx-dev.yqtz8.com/vendor/yiisoft/yii2/base/Application.php(375): yii\web\Application->handleRequest(Object(yii\web\Request))
#14 /niub/www/sourcecode/wx-dev.yqtz8.com/frontend/web/index.php(20): yii\base\Application->run()
#15 {main}


------------ 2016-10-27 16:03:03
?? 
Yii::$app->user->returnUrl


----------- 2016-10-19 17:18:02

   public function getMission()
    {
        if ($this->mission_type == MissionType::PERIOD) {
            return $this->hasOne(PeriodMission::className(), ['id' => 'mission_id']);
        } else {
            return $this->hasOne(Mission::className(), ['id' => 'mission_id']);
        }
    }

    not sure why, hasOne returns below object instead of a raw mission.
{
    "sql": null,
    "on": null,
    "joinWith": null,
    "select": null,
    "selectOption": null,
    "distinct": null,
    "from": null,
    "groupBy": null,
    "join": null,
    "having": null,
    "union": null,
    "params": [],
    "where": null,
    "limit": null,
    "offset": null,
    "orderBy": null,
    "indexBy": null,
    "modelClass": "frontend\\models\\Mission",
    "with": null,
    "asArray": null,
    "multiple": false,
    "primaryModel": {
        "id": 21963,
        "mission_id": 2134,
        "user_id": 29,
        "regist_time": "2016-09-19 13:22:58",
        "status": 0,
        "checkin_valid": null,
        "checkin_invalid": null,
        "checkin_count": 0,
        "mission_type": 0,
        "start_round": 1,
        "invite_image": null,
        "inviter_id": 0
    },
    "link": {
        "id": "mission_id"
    },
    "via": null,
    "inverseOf": null
}


-------- 2016-10-12 10:54:46

                <div class="status text-center">
                  <span class="block" v-if="score.reward!=null" v-text="(score.reward / 100.0) | currency '¥' 2"></span>
                  <p v-if="score.reward!=null">奖励</p>
                  <p v-if="score.reward==null">未领奖</p>
                </div>
                
------ 2016-10-10 19:02:10
from local:
 when "php yii console/start-worker" is in process, modifying card_jobs will be seen by the thread. 
 however in online test machine, modifying the card_jobs' event_time is not seen until re-deployment from jenkins.

------------ 2016-10-10 15:33:01

$ranking = array_slice($ranking, $offset, $count, true);
  be careful with array_slice if the key is a number. 

 * @param bool $preserve_keys [optional] <p>
 * Note that array_slice will reorder and reset the
 * array indices by default. You can change this behaviour by setting
 * preserve_keys to true.

----------- 2016-10-10 14:24:16
mysql> select * from card_base where mission_id=201;
+-----+------+--------+---------------------+-----------------+------------+------------------------+--------------+---------+---------------------+
| id  | type | obj_id | create_time         | title           | mission_id | description            | mission_type | user_id | update_time         |
+-----+------+--------+---------------------+-----------------+------------+------------------------+--------------+---------+---------------------+
| 443 |    2 |    518 | 2016-09-30 19:48:09 | NULL            |        201 | 第一周第1次打卡        |            1 |      12 | 2016-09-30 19:48:22 |
| 444 |    2 |    519 | 2016-09-30 19:48:09 | NULL            |        201 | 第一周第1次打卡        |            1 |       3 | 2016-09-30 19:48:39 |
| 447 |    2 |    522 | 2016-09-30 20:09:09 | NULL            |        201 | 第二周第1次打卡        |            1 |      12 | 2016-09-30 20:09:27 |
| 448 |    2 |    523 | 2016-09-30 20:10:09 | NULL            |        201 | 第二周第1次打卡        |            1 |       2 | 2016-09-30 20:10:18 |
| 522 |    1 |    163 | 2016-09-23 04:00:00 | 第一周战报      |        201 |                        |            1 |       0 | 2016-10-08 13:00:42 |
| 545 |    1 |    168 | 2016-09-30 04:00:00 | 第二周战报      |        201 |                        |            1 |       0 | 2016-10-10 13:20:27 |
+-----+------+--------+---------------------+-----------------+------------+------------------------+--------------+---------+---------------------+
6 rows in set (0.00 sec)

mysql> select * from checkin_in where mission_id=201;
ERROR 1146 (42S02): Table 'tz8_dev.checkin_in' doesn't exist
mysql> select * from checkin where mission_id=201;
+-----+--------------+------------+--------------+---------+------------------------------------------------------------------+------------+--------------+---------------+---------------------+---------------+
| id  | challenge_id | mission_id | mission_type | user_id | photo                                                            | like_count | unlike_count | comment_count | checkin_time        | veto_admin_id |
+-----+--------------+------------+--------------+---------+------------------------------------------------------------------+------------+--------------+---------------+---------------------+---------------+
| 518 |          920 |        201 |            1 |      12 | https://qiniu.yqtz8.com/images/2016_09_30_19_48_22_57ee5106394bd |          1 |            1 |             0 | 2016-09-17 05:00:00 |            -1 |
| 519 |          921 |        201 |            1 |       3 | https://qiniu.yqtz8.com/images/2016_09_30_19_48_38_57ee5116ea75f |          1 |            0 |             0 | 2016-09-17 05:00:00 |            12 |
| 522 |          920 |        201 |            1 |      12 | https://qiniu.yqtz8.com/images/2016_09_30_20_09_27_57ee55f7bf7c6 |          1 |            0 |             0 | 2016-09-24 05:00:00 |            -1 |
| 523 |          922 |        201 |            1 |       2 | https://qiniu.yqtz8.com/images/2016_09_30_20_10_18_57ee562a37a0b |          1 |            0 |             0 | 2016-09-17 05:00:00 |            -1 |
+-----+--------------+------------+--------------+---------+------------------------------------------------------------------+------------+--------------+---------------+---------------------+---------------+
4 rows in set (0.00 sec)


---------- 2016-10-10 13:26:58
持续2周的挑战，没有第二周周报挑战结束: 任务实际结束时间是10.14，今天是10.10，可见是因为手动修改数据库导致任务提前结束，生成周报的任务时间是创建任务的时候确定的，手动修改数据库并不会触发任务执行，因此如果有测试需求的话需要强行执行一次生成周报的任务「通过修改card_jobs表中相应任务的event_time」。

------------ 2016-10-09 17:23:18

actionReport
  getChallengeReport will fail if reward is not confirmed.

2016-10-09 17:20:46 [187][error][abc\frontend\components\ErrorHandler::logException] frontend\components\AppException: 无挑战战报 in /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/ReportManager.php:251
Stack trace:
#0 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/controllers/MissionController.php(902): frontend\components\ReportManager::getChallengeReport(Object(frontend\models\PeriodMission), Object(frontend\models\User))
#1 [internal function]: frontend\controllers\MissionController->actionReport('118', '1', NULL)
#2 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/InlineAction.php(55): call_user_func_array(Array, Array)
#3 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Controller.php(154): yii\base\InlineAction->runWithParams(Array)
#4 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Module.php(454): yii\base\Controller->runAction('report', Array)
#5 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/web/Application.php(87): yii\base\Module->runAction('mission/report', Array)
#6 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Application.php(375): yii\web\Application->handleRequest(Object(yii\web\Request))
#7 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/web/index.php(21): yii\base\Application->run()
#8 {main}
    in /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/util/Logger.php:29
    in /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/ErrorHandler.php:25
2016-10-09 17:22:49 [-][error][abc\frontend\components\ErrorHandler::logException] yii\base\InvalidRouteException: Unable to resolve the request: mission/118 in /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Controller.php:125
Stack trace:
#0 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Module.php(454): yii\base\Controller->runAction('118', Array)
#1 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/web/Application.php(87): yii\base\Module->runAction('mission/118', Array)
#2 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Application.php(375): yii\web\Application->handleRequest(Object(yii\web\Request))
#3 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/web/index.php(21): yii\base\Application->run()
#4 {main}

/home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/ReportManager.php
        // 有奖金尚未领取，视为挑战未结束，不生成战报
        $rewards = ChallengeReward::getRewardsByChallenge($challenge->id, null);
        foreach ($rewards as $reward) {
            if ($reward->status == RewardStatus::PENDING) {
                // 当前挑战有奖金尚未领取，不生成战报
                Logger::info('test 3');
                return;
            }
        }

------------ 2016-10-09 11:02:08
2016-10-08 22:35:09 [57][error][abc\frontend\components\ErrorHandler::logException] exception 'frontend\components\AppException' with message 'invalid date' in /niub/www/sourcecode/wx.yqtz8.com/frontend/components/util/TwigHelper.php:522
Stack trace:
#0 /niub/www/sourcecode/wx.yqtz8.com/frontend/components/event/MessageCardBase.php(45): frontend\components\util\TwigHelper::getFriendlyTimeText('2016-10-08 22:3...')
#1 /niub/www/sourcecode/wx.yqtz8.com/frontend/components/event/MessageCardCheckin.php(76): frontend\components\event\MessageCardBase->loadFromBase(Object(frontend\models\event\CardBase))
#2 /niub/www/sourcecode/wx.yqtz8.com/frontend/components/DynamicCardsManager.php(82): frontend\components\event\MessageCardCheckin::generateMessageForPage(Object(frontend\models\event\CardBase))
#3 /niub/www/sourcecode/wx.yqtz8.com/frontend/controllers/ajax/MissionController.php(893): frontend\components\DynamicCardsManager::getMissionCardsByType('292', '0', 2, '0', '10')
#4 [internal function]: frontend\controllers\ajax\MissionController->actionDynamicCards('292', '0', '0', '10')
#5 /niub/www/sourcecode/wx.yqtz8.com/vendor/yiisoft/yii2/base/InlineAction.php(55): call_user_func_array(Array, Array)
#6 /niub/www/sourcecode/wx.yqtz8.com/vendor/yiisoft/yii2/base/Controller.php(154): yii\base\InlineAction->runWithParams(Array)
#7 /niub/www/sourcecode/wx.yqtz8.com/vendor/yiisoft/yii2/base/Module.php(454): yii\base\Controller->runAction('dynamic-cards', Array)
#8 /niub/www/sourcecode/wx.yqtz8.com/vendor/yiisoft/yii2/web/Application.php(87): yii\base\Module->runAction('ajax/mission/dy...', Array)
#9 /niub/www/sourcecode/wx.yqtz8.com/vendor/yiisoft/yii2/base/Application.php(375): yii\web\Application->handleRequest(Object(yii\web\Request))
#10 /niub/www/sourcecode/wx.yqtz8.com/frontend/web/index.php(20): yii\base\Application->run()
#11 {main}

mysql> select * from checkin where mission_id=292 and mission_type=0;
+-----+--------------+------------+--------------+---------+------------------------------------------------------------------+------------+--------------+---------------+---------------------+---------------+
| id  | challenge_id | mission_id | mission_type | user_id | photo                                                            | like_count | unlike_count | comment_count | checkin_time        | veto_admin_id |
+-----+--------------+------------+--------------+---------+------------------------------------------------------------------+------------+--------------+---------------+---------------------+---------------+
| 666 |          585 |        292 |            0 |       9 | https://qiniu.yqtz8.com/images/2016_10_08_22_27_45_57f902619902c |          1 |            0 |             0 | 2016-10-08 22:27:10 |            -1 |
| 667 |          586 |        292 |            0 |      57 | https://qiniu.yqtz8.com/images/2016_10_08_22_35_06_57f9041ab2689 |          1 |            0 |             0 | 2016-10-08 22:35:10 |            -1 |
| 668 |          583 |        292 |            0 |       4 | https://qiniu.yqtz8.com/images/2016_10_08_22_35_06_57f9041ac029c |          1 |            0 |             0 | 2016-10-08 22:35:10 |            -1 |
| 671 |          581 |        292 |            0 |       8 | https://qiniu.yqtz8.com/images/2016_10_09_08_48_26_57f993da1d8a4 |          1 |            0 |             0 | 2016-10-09 08:48:10 |            -1 |
+-----+--------------+------------+--------------+---------+------------------------------------------------------------------+------------+--------------+---------------+---------------------+---------------+
4 rows in set (0.00 sec)


--------- 2016-10-09 00:31:31
why git push pushed to all remote branches?

where is php log by default?
----------- 2016-09-29 21:58:21

    public static function getVoteUserList($cid, $vote = 1)
    {
        $list = (new \yii\db\Query())
            ->select('user.id, user.nickname, user.head_img, checkin_vote.create_at')
            ->from('checkin_vote')
            ->innerJoin('user', 'checkin_vote.user_id = user.id')
            ->where([
                    'checkin_vote.checkin_id' => $cid,
                    'checkin_vote.opt' => $vote,
                ])
            ->indexBy('id')
            ->all();
        return $list;
    }

sample output: ?? why "10" is the key?? indexBy('id') ??
  {"10":{"id":"10","nickname":"叶安华","head_img":"http://wx.qlogo.cn/mmopen/PiajxSqBRaEJkp2ibNelHjpXwSUzqSaUIVRJ8Lxd9HtLDxMic2ziasKXY52oW5RLNLEH203qBoXUiavuZsp0aDnGGtQ/0","create_at":"2016-09-29 21:48:04"}}

------------ 2016-09-29 18:40:31
twig:


                      <span class="time text-left">{{ checkin.checkin_time | html }}</span>

                    <p class="weui_media_desc text-left">{{checkin.checkin_time | date('Y-m-d H:i')}}</p>



----------- 2016-09-29 15:39:05
2016-09-29 15:36:20 [187][error][abc\frontend\components\ErrorHandler::logException] frontend\components\AppException: action post 不再使用 in /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/controllers/MissionController.php:196
Stack trace:
#0 [internal function]: frontend\controllers\MissionController->actionIndex('368', NULL, 0, NULL)
#1 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/InlineAction.php(55): call_user_func_array(Array, Array)
#2 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Controller.php(154): yii\base\InlineAction->runWithParams(Array)
#3 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Module.php(454): yii\base\Controller->runAction('index', Array)
#4 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/web/Application.php(87): yii\base\Module->runAction('mission/index', Array)
#5 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Application.php(375): yii\web\Application->handleRequest(Object(yii\web\Request))
#6 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/web/index.php(21): yii\base\Application->run()
#7 {main}
    in /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/util/Logger.php:29
    in /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/ErrorHandler.php:25





--------- 2016-09-28 18:50:44
do not throw app exception in this case of null mission.

$ php yii tools/create-history-cards

 to generate cards for checkins: 1 Exception 'frontend\components\AppException' with message '找不到任务'

in /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/util/CommonHelpers.php:13

Stack trace:
#0 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/CheckinManager.php(109): frontend\components\util\CommonHelpers::throwExceptionIfEmpty(NULL, '\xE6\x89\xBE\xE4\xB8\x8D\xE5\x88\xB0\xE4\xBB\xBB\xE5\x8A\xA1')
#1 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/event/MessageCardCheckin.php(32): frontend\components\CheckinManager::getCardCheckinDescription(1)
#2 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/event/CardEventManager.php(60): frontend\components\event\MessageCardCheckin::generateMessageForDb(1)
#3 [internal function]: frontend\components\event\CardEventManager::handleGenerateCard(Object(frontend\components\event\CardEvent))
#4 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Component.php(541): call_user_func(Array, Object(frontend\components\event\CardEvent))
#5 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/event/CardEventManager.php(20): yii\base\Component->trigger('generateCard', Object(frontend\components\event\CardEvent))
#6 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/DynamicCardsManager.php(21): frontend\components\event\CardEventManager::sendEvent('generateCard', Object(frontend\components\event\MessageCardCheckin))
#7 /home/kidd/workspace/bc/advanced/MissionImpossible/console/controllers/ToolsController.php(187): frontend\components\DynamicCardsManager::generateCardCheckin(1)
#8 [internal function]: console\controllers\ToolsController->actionCreateHistoryCards()
#9 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/InlineAction.php(55): call_user_func_array(Array, Array)
#10 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Controller.php(154): yii\base\InlineAction->runWithParams(Array)
#11 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/console/Controller.php(119): yii\base\Controller->runAction('create-history-...', Array)
#12 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Module.php(454): yii\console\Controller->runAction('create-history-...', Array)
#13 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('tools/create-hi...', Array)
#14 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/console/Application.php(147): yii\console\Application->runAction('tools/create-hi...', Array)
#15 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Application.php(375): yii\console\Application->handleRequest(Object(yii\console\Request))
#16 /home/kidd/workspace/bc/advanced/MissionImpossible/yii(27): yii\base\Application->run()
#17 {main}


------------ 2016-09-28 18:31:17
when triggered from console task: handleGenerateCard is not triggered

    public static function sendEvent($event_type, $message)
    {
        echo 'he';
        $event = new CardEvent;
        $event->message = $message;
        Yii::$app->trigger($event_type, $event);
        echo ',';
    }

    /**
     * handle card generation event and save card to db
     * @param $event
     * @return bool
     */
    public static function handleGenerateCard($event)
    {
        echo 's';

fix: /home/kidd/workspace/bc/advanced/MissionImpossible/console/config/main.php

$config = [
    'id' => 'app-console',
    'language' => 'zh-CN',
    'timeZone' => 'Asia/Shanghai',
    'basePath' => dirname(__DIR__),
    'bootstrap' => [
        'log',
        'frontend\components\event\EventBootstrap',
    ],

----------- 2016-09-20 20:11:38
~/workspace/bc/advanced/MissionImpossible
$ rm -rf /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/runtime/Twig/cache

2016-09-20 20:10:51 [187][error][abc\frontend\components\ErrorHandler::logException] Error: Call to undefined method frontend\components\util\TwigHelper::format_time_range() in /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/runtime/Twig/cache/5c/5ce9787304024c41606a987e5faf8b4383fa9206b184f786fc3ce97eabffa0f4.php:31
Stack trace:
#0 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/twig/twig/lib/Twig/Template.php(387): __TwigTemplate_40dddb1ce9c3a616ff19d254cbffbd5cb6f0ade4eb77260f29ec123cb6b187c3->doDisplay(Array, Array)
#1 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/twig/twig/lib/Twig/Template.php(355): Twig_Template->displayWithErrorHandling(Array, Array)
#2 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/runtime/Twig/cache/86/866517e73c5cae94f433c30073ac440d98c58c8b54933633dc688d043da345d5.php(87): Twig_Template->display(Array)
#3 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/twig/twig/lib/Twig/Template.php(387): __TwigTemplate_ac4d84c0c9079877e449e53d29fe083c9397bf1efbbcb58c437be003296b9247->doDisplay(Array, Array)
#4 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/twig/twig/lib/Twig/Template.php(355): Twig_Template->displayWithErrorHandling(Array, Array)
#5 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/twig/twig/lib/Twig/Template.php(366): Twig_Template->display(Array)
#6 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/twig/twig/lib/Twig/Environment.php(347): Twig_Template->render(Array)
#7 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2-twig/ViewRenderer.php(152): Twig_Environment->render('pre-checkin_new...', Array)
#8 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/View.php(245): yii\twig\ViewRenderer->render(Object(yii\web\View), '/home/kidd/work...', Array)
#9 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/View.php(149): yii\base\View->renderFile('/home/kidd/work...', Array, Object(frontend\controllers\MissionController))
#10 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Controller.php(378): yii\base\View->render('single/pre-chec...', Array, Object(frontend\controllers\MissionController))
#11 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/controllers/WebController.php(90): yii\base\Controller->render('single/pre-chec...', Array, false)
#12 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/controllers/MissionController.php(156): frontend\controllers\WebController->render('single/pre-chec...', Array)
#13 [internal function]: frontend\controllers\MissionController->actionIndex('2220', NULL, '0', NULL)
#14 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/InlineAction.php(55): call_user_func_array(Array, Array)
#15 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Controller.php(154): yii\base\InlineAction->runWithParams(Array)
#16 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Module.php(454): yii\base\Controller->runAction('index', Array)
#17 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/web/Application.php(87): yii\base\Module->runAction('mission/index/', Array)
#18 /home/kidd/workspace/bc/advanced/MissionImpossible/vendor/yiisoft/yii2/base/Application.php(375): yii\web\Application->handleRequest(Object(yii\web\Request))
#19 /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/web/index.php(21): yii\base\Application->run()
#20 {main}
    in /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/util/Logger.php:29
    in /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/components/ErrorHandler.php:25


------------ 2016-09-20 18:24:12
2016-09-20 18:23:40 [][info][abc\frontend\controllers\LoginController::actionLogin] account is null with db: {"dsn":"mysql:host=10.11.255.110;port=31306;dbname=tz8","username":"tz8","password":"7fedb3187f03","attributes":null,"pdo":{},"enableSchemaCache":false,"schemaCacheDuration":3600,"schemaCacheExclude":[],"schemaCache":"cache","enableQueryCache":true,"queryCacheDuration":3600,"queryCache":"cache","charset":"utf8mb4","emulatePrepare":null,"tablePrefix":"admin_","schemaMap":{"pgsql":"yii\\db\\pgsql\\Schema","mysqli":"yii\\db\\mysql\\Schema","mysql":"yii\\db\\mysql\\Schema","sqlite":"yii\\db\\sqlite\\Schema","sqlite2":"yii\\db\\sqlite\\Schema","sqlsrv":"yii\\db\\mssql\\Schema","oci":"yii\\db\\oci\\Schema","mssql":"yii\\db\\mssql\\Schema","dblib":"yii\\db\\mssql\\Schema","cubrid":"yii\\db\\cubrid\\Schema"},"pdoClass":null,"commandClass":"yii\\db\\Command","enableSavepoint":true,"serverStatusCache":"cache","serverRetryInterval":600,"enableSlaves":true,"slaves":[],"slaveConfig":[],"masters":[],"masterConfig":[]}

----------- 2016-09-11 23:00:06
met an issue: 
  accidentally put a line into Application.php, and the page does not show from apache. and the http response is connection closed, internal error.
  Luckily noticed the warnings in the phpstorm project view warnings and saw some warnings in Application.php, and then removed the accidentally pasted line.


---------- 2016-09-13 14:01:28
http://www.baidu.com/link?url=_J_taS2tnMFTvSMuq8H3TZOJtKfqDw2gSSDuGGwCFaECsbqDslE5QlmsLKaXpqf_jxT8bb_zobQLtNBIE1LEw-CIxVbff35jYyLPNzMuajO&wd=&eqid=ee3d2256000048730000000257d795ed
 发生了Post错误:错误代码40005,微信返回错误信息:invalid file type
给客户部署 PxxCms, 使用群发功能发送图文的的时候提示: 发生了Post错误:错误代码40005,微信返回错误信息:invalid file type, 没学过php伤不起 ...
Google 到某哥的博客, 有解决方案, 但是tmd 收费...擦了, 自己改
找到 \PigCms\Lib\Action\User\MessageAction.class.php 的 sendAll 方法, 大约112 行, 改成如下,注释掉第一行file_put_contents(...);
//file_put_contents(CONF_PATH.'img_'.$img['id'].'.jpg',file_get_contents($img['pic']));
$postMedia['media']=$_SERVER['DOCUMENT_ROOT'].str_replace('http://'.$_SERVER['HTTP_HOST'],'',$img['pic']);
原因看代码应该就理解了.
转载请注明出处:http://duwei.cnblogs.com/

----------- 2016-09-01 18:21:01
headache:
  used "DEFAULT" as a var name, and php stopped without any warning. But phpstorm has some warning as seen from ide, which i did not notice.


--------- 2016-09-01 17:14:05

2016-09-01 17:12:04 [][info][abc\frontend\components\wxmp\MpManager::processMessage] no handler found
2016-09-01 17:12:04 [][info][abc\frontend\components\wxmp\MpManager::processMessage] no handler found
2016-09-01 17:12:05 [][info][abc\frontend\components\wxmp\MpManager::processMessage] no handler found
2016-09-01 17:12:06 [][info][abc\frontend\controllers\LoginController::getWechatLoginUrl] wechat login callback: https://tz8-dev.muder.pw/login/wechat-login-callback
2016-09-01 17:12:06 [-][info][abc\frontend\controllers\LoginController::actionWechatLoginCallback] used app id: wx7a84fcbe9e98e97f
2016-09-01 17:12:06 [-][info][abc\frontend\components\AccountManager::handleWechatLoginCallback] wechat login callback, code: 0412Tsab1IrlVq0G9Kab1siqab12TsaL, appId: wx7a84fcbe9e98e97f
2016-09-01 17:12:06 [-][info][abc\frontend\components\AccountManager::loginWithWechat] Call function with args: ["0412Tsab1IrlVq0G9Kab1siqab12TsaL","wx7a84fcbe9e98e97f"]
2016-09-01 17:12:06 [-][info][abc\frontend\components\AccountManager::loginWithWechat] Use wechat user session
2016-09-01 17:12:06 [-][info][abc\frontend\components\AccountManager::loginWithWechat] get account by openId: ook-7wLch8xSS-mM-urjLCCs91n0 unionId: ook-7wLch8xSS-mM-urjLCCs91n0
2016-09-01 17:12:06 [-][info][abc\frontend\components\AccountManager::loginWithWechat] account is empty, make a new ook-7wLch8xSS-mM-urjLCCs91n0 unionId: ook-7wLch8xSS-mM-urjLCCs91n0
2016-09-01 17:12:13 [][info][abc\frontend\components\wxmp\MpManager::processMessage] no handler found



------------ 2016-08-30 21:06:56
Unable to find 'app\assets\AppAsset' in file: /home/kidd/workspace/bc/advanced/MissionImpossible/frontend/assets/AppAsset.php. Namespace missing?

/home/kidd/workspace/bc/advanced/MissionImpossible/frontend/views/layouts/main.php
use app\assets\AppAsset;
=>
use frontend\assets\AppAsset;

------------- 2016-08-25 20:19:06

~/workspace/bc/advanced/MissionImpossible
$ ./init 
Yii Application Initialization Tool v1.0

Which environment do you want the application to be initialized in?

  [0] Development
  [1] Test
  [2] Production

  Your choice [0-2, or "q" to quit] 0

  Initialize the application under 'Development' environment? [yes|no] yes

  Start initialization ...

  unchanged yii
  unchanged console/config/params-local.php
  unchanged console/config/main-local.php
  unchanged common/config/params-local.php
  unchanged common/config/main-local.php
  unchanged frontend/config/params-local.php
      exist frontend/config/main-local.php
            ...overwrite? [Yes|No|All|Quit] No
       skip frontend/config/main-local.php
  unchanged frontend/web/index-test.php
   generate frontend/web/index.php
  unchanged tests/codeception/config/config-local.php
  unchanged backend/config/params-local.php
      exist backend/config/main-local.php
            ...overwrite? [Yes|No|All|Quit] No
       skip backend/config/main-local.php
  unchanged backend/web/index-test.php
  unchanged backend/web/index.php
   generate cookie validation key in backend/config/main-local.php
   generate cookie validation key in frontend/config/main-local.php
      chmod 0777 backend/runtime
      chmod 0777 backend/web/assets
      chmod 0777 frontend/runtime
      chmod 0777 frontend/web/assets
      chmod 0755 yii
      chmod 0755 tests/codeception/bin/yii

  ... initialization completed.



------------- 2016-08-23 15:44:18

2016-08-23 15:41:40 [187][error][abc\app\controllers\ajax\MissionController::actionCreate] save mission failed: eewer
    in /home/kidd/workspace/bc/MissionImpossible/components/util/Logger.php:25
    in /home/kidd/workspace/bc/MissionImpossible/controllers/ajax/MissionController.php:106

post on failure:
{
    "mt": "1",
    "PeriodMission": {
        "announce": "eewer",
        "checkin_count": "1",
        "ticket": "100",
        "start_time": "2016/08/23",
        "repeat": "1",
        "checkin_type": "1",
        "checkin_unlike_invalid": "50",
        "checkin_like_valid": "0",
        "description": ""
    }
}

post data on success:
{
    "mt": "1",
    "PeriodMission": {
        "announce": "eewer",
        "checkin_count": "1",
        "ticket": "100",
        "start_time": "2016/08/23",
        "repeat": "1",
        "checkin_type": "1",
        "checkin_unlike_invalid": "50",
        "checkin_like_valid": "0",
        "join_after_start": "1",
        "description": ""
    }
}

if join_after_start of an active record is set to false, the corresponding value would be 'false', and save will fail, either pass in 0 or '0'.

| join_after_start       | tinyint(1)     | YES  |     | 1       |                |

{"mt":"1","PeriodMission":{"announce":"adsf","checkin_count":"1","ticket":"100","start_time":"2016/08/23","repeat":"1","checkin_type":"1","checkin_unlike_invalid":"50","checkin_like_valid":"0","description":""}}

{"join_after_start":false,"template_id":1,"announce":"adsf","name":"adsf","period":7,"min_unit":1,"checkin_time":"","checkin_count":"1","checkin_unlike_invalid":"50","checkin_like_valid":"0","repeat":"1","creator_id":4,"create_time":{"expression":"NOW()","params":[]},"start_time":"2016/08/23","ticket":"100","checkin_type":"1","description":"","reward_total":0,"joined_count":0}

Logger::error(Json::encode($mission->getErrors()));
2016-08-23 16:24:24 [187][error][abc\app\controllers\ajax\MissionController::actionCreate] {"join_after_start":["Join After Start必须是整数。"]}
    in /home/kidd/workspace/bc/MissionImpossible/components/util/Logger.php:25
    in /home/kidd/workspace/bc/MissionImpossible/controllers/ajax/MissionController.php:106

    
------------ 2016-08-23 10:44:12
potential mission create and join failure:

canJoin() failure: [because the web server time might be earlier than the db server time]
                $mission->create_time = date('Y-m-d H:i:s', time()); // new Expression('NOW()');

update the system time:
$ which ntpdate
/usr/sbin/ntpdate
kidd@kidd-OptiPlex-9020:
~/workspace/php
$ ntpdate cn.pool.ntp.org
23 Aug 10:51:17 ntpdate[462]: bind() fails: Permission denied
kidd@kidd-OptiPlex-9020:
~/workspace/php
$ sudo ntpdate cn.pool.ntp.org
[sudo] password for kidd: 
23 Aug 10:51:45 ntpdate[791]: step time server 202.118.1.81 offset 16.206781 sec
3.  将系统时间写入硬件时间# hwclock --systohc
-------------- 2016-08-23 10:26:29
/home/kidd/workspace/bc/MissionImpossible/config/web.php
  'log' => require(__DIR__ . '/log.php'),
              'logFile' => '@runtime/logs/framework.log', [/home/kidd/workspace/bc/MissionImpossible/config/log.php]



below error is seen from browser debug's http response, but there is no any output from server side.

{"success":true,"message":"","data":{"name":"Exception","message":"Call to undefined method app\\components\\util\\Logger::inof()","code":0,"type":"Error","file":"/home/kidd/workspace/bc/MissionImpossible/models/PeriodMission.php","line":142,"stack-trace":["#0 /home/kidd/workspace/bc/MissionImpossible/components/MissionManager.php(61): app\\models\\PeriodMission->canJoin()","#1 /home/kidd/workspace/bc/MissionImpossible/controllers/ajax/MissionController.php(77): app\\components\\MissionManager::joinMission(1782, '1')","#2 [internal function]: app\\controllers\\ajax\\MissionController->actionCreate()","#3 /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/base/InlineAction.php(55): call_user_func_array(Array, Array)","#4 /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/base/Controller.php(154): yii\\base\\InlineAction->runWithParams(Array)","#5 /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/base/Module.php(454): yii\\base\\Controller->runAction('create', Array)","#6 /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/web/Application.php(87): yii\\base\\Module->runAction('ajax/mission/cr...', Array)","#7 /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/base/Application.php(375): yii\\web\\Application->handleRequest(Object(yii\\web\\Request))","#8 /home/kidd/workspace/bc/MissionImpossible/web/index.php(17): yii\\base\\Application->run()","#9 {main}"]}}


!!!!!!!! refer to /home/kidd/workspace/bc/MissionImpossible/runtime/logs/framework.log
sample log:
2016-08-23 10:29:06 [187][error][Error] Error: Call to undefined method app\components\util\Logger::inof() in /home/kidd/workspace/bc/MissionImpossible/models/PeriodMission.php:142
Stack trace:
#0 /home/kidd/workspace/bc/MissionImpossible/components/MissionManager.php(61): app\models\PeriodMission->canJoin()
#1 /home/kidd/workspace/bc/MissionImpossible/controllers/ajax/MissionController.php(77): app\components\MissionManager::joinMission(1783, '1')
#2 [internal function]: app\controllers\ajax\MissionController->actionCreate()
#3 /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/base/InlineAction.php(55): call_user_func_array(Array, Array)
#4 /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/base/Controller.php(154): yii\base\InlineAction->runWithParams(Array)
#5 /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/base/Module.php(454): yii\base\Controller->runAction('create', Array)
#6 /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/web/Application.php(87): yii\base\Module->runAction('ajax/mission/cr...', Array)
#7 /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/base/Application.php(375): yii\web\Application->handleRequest(Object(yii\web\Request))
#8 /home/kidd/workspace/bc/MissionImpossible/web/index.php(17): yii\base\Application->run()
#9 {main}
2016-08-23 10:29:06 [187][info][application] $_POST = [
    'mt' => '1'
    'PeriodMission' => [
        'announce' => 'sdf'
        'checkin_count' => '1'
        'ticket' => '100'
        'start_time' => '2016/08/23'
        'repeat' => '1'
        'checkin_type' => '1'
        'checkin_unlike_invalid' => '50'
        'checkin_like_valid' => '0'
        'join_after_start' => '1'
        'description' => ''
    ]
]

$_COOKIE = [
    'PHPSESSID' => '4h6nd9jugag9p49qncsiercqv3'
    'userId' => '187'
    '_identity' => '[\"187\",\"i-rgIhN9yrgLPQygIppTr0Q3BWnjc2x2\",2592000]'
]

$_SESSION = [
    '__flash' => []
    '__page_session' => []
    '__returnUrl' => '/mission/create?mt=1'
    '__id' => '187'
]

$_SERVER = [
    'DOCUMENT_ROOT' => '/home/kidd/workspace/bc/MissionImpossible/web'
    'REMOTE_ADDR' => '10.12.0.1'
    'REMOTE_PORT' => '45731'
    'SERVER_SOFTWARE' => 'PHP 7.0.9-1+deb.sury.org~trusty+1 Development Server'
    'SERVER_PROTOCOL' => 'HTTP/1.1'
    'SERVER_NAME' => '10.12.0.1'
    'SERVER_PORT' => '8080'
    'REQUEST_URI' => '/ajax/mission/create'
    'REQUEST_METHOD' => 'POST'
    'SCRIPT_NAME' => '/index.php'
    'SCRIPT_FILENAME' => '/home/kidd/workspace/bc/MissionImpossible/web/index.php'
    'PATH_INFO' => '/ajax/mission/create'
    'PHP_SELF' => '/index.php/ajax/mission/create'
    'HTTP_HOST' => '10.12.0.1:8080'
    'HTTP_CONNECTION' => 'keep-alive'
    'CONTENT_LENGTH' => '363'
    'HTTP_CONTENT_LENGTH' => '363'
    'HTTP_ACCEPT' => '*/*'
    'HTTP_ORIGIN' => 'http://10.12.0.1:8080'
    'HTTP_X_REQUESTED_WITH' => 'XMLHttpRequest'
    'HTTP_USER_AGENT' => 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.89 Safari/537.36'
    'CONTENT_TYPE' => 'application/x-www-form-urlencoded; charset=UTF-8'
    'HTTP_CONTENT_TYPE' => 'application/x-www-form-urlencoded; charset=UTF-8'
    'HTTP_REFERER' => 'http://10.12.0.1:8080/mission/create?mt=1'
    'HTTP_ACCEPT_ENCODING' => 'gzip, deflate'
    'HTTP_ACCEPT_LANGUAGE' => 'en-US,en;q=0.8,zh-CN;q=0.6,zh;q=0.4'
    'HTTP_COOKIE' => 'PHPSESSID=4h6nd9jugag9p49qncsiercqv3; userId=187; _identity=%5B%22187%22%2C%22i-rgIhN9yrgLPQygIppTr0Q3BWnjc2x2%22%2C2592000%5D'
    'REQUEST_TIME_FLOAT' => 1471919346.1335
    'REQUEST_TIME' => 1471919346
]

----------- 2016-08-19 20:56:03

how to define a yii class member: const array


------------ 2016-08-19 18:21:28
/home/kidd/workspace/bc/MissionImpossible/models/Mission.php
    /*
     * 挑战成功用户
     * */
    public function getSuccessors()
    {
        $challenges = $this->challenges;

        $users = array();
        foreach ($challenges as $challenge) {
            if ($challenge->isSucceed()) {
                $users[] = $challenge->user;
            }
        }

        return $users;
    }

/home/kidd/workspace/bc/MissionImpossible/views/mission/single/my-redpacket.html 
  <!-- 好友红包列表 -->
  <div class="panel checked-wrapper">
    <div class="hd title app-default-bgcolor">
      <span class="page_title">已领取 {{reward_send_count}}/{{reward_total_count}} 个，共{{mission.reward_send  / 100.0}}/{{mission.reward_total  / 100.0}} 元</h1>
    </div>
    <div class="bd">
      <div class="weui_panel weui_panel_access">
        <div class="weui_panel_bd">
            <ul class="mem-list text-left">
                {% for redpack in redPackList %}
                <li class="weui_media_box weui_media_appmsg">
                <div class="weui_media_hd">
                    <img class="weui_media_appmsg_thumb" src="{{redpack.head_img}}" alt="">
                </div>
                <div class="weui_media_bd">
                    <h4 class="weui_media_title">{{redpack.username}}</h4>
                    <p class="weui_media_desc">{{redpack.send_time | date('H:i')}}</p>
                </div>
                <span class="time text-right">{{redpack.money/100.0}} 元</span>
                </li>
                {% endfor %}
            </ul>
        </div>


------------ 2016-08-19 14:49:49
public static function getCheckinDesc($mid, $round=null, $uid=null)
  get round failed, you need to pass $round if mission not ongoing right now



---------- 2016-08-18 20:42:46
Database Exception – yii\db\Exception
SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '== TO_DAYS(NOW())' at line 1
The SQL being executed was: SELECT COUNT(*) FROM `checkin` WHERE checkin.challenge_id = 8629 and TO_DAYS(checkin_time) == TO_DAYS(NOW())

Error Info: Array
(
    [0] => 42000
    [1] => 1064
    [2] => You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '== TO_DAYS(NOW())' at line 1
)

↵
Caused by: PDOException
SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '== TO_DAYS(NOW())' at line 1

in /home/kidd/workspace/bc/MissionImpossible/vendor/yiisoft/yii2/db/Command.php at line 900

    public function getCheckinCountInUnit()
    {
        // 单次打卡任务
        if ($this->mission_type == MissionType::SINGLE){
            return ($this->checkinCount);
        } elseif ( $this->mission_type == MissionType::PERIOD ) {
        // 周期打卡任务
            if ($this->mission->min_unit == CheckinUnit::HOUR) {
                return  (new \yii\db\Query())
                    ->from('checkin')
                    ->where("checkin.challenge_id = {$this->id} and TO_DAYS(checkin_time) == TO_DAYS(NOW()) and HOUR(checkin_time) ==  HOUR(NOW())")
                    ->count();
            } elseif ($this->mission->min_unit == CheckinUnit::DAY) {
                return  (new \yii\db\Query())
                    ->from('checkin')
                    ->where("checkin.challenge_id = {$this->id} and TO_DAYS(checkin_time) == TO_DAYS(NOW())")
                    ->count();
            } elseif ($this->mission->min_unit == CheckinUnit::WEEK) {
                return  (new \yii\db\Query())
                    ->from('checkin')
                    ->where("checkin.challenge_id = {$this->id} and WEEKOFYEAR(checkin_time) == WEEKOFYEAR(NOW())")
                    ->count();
            }
        }

        return 0;
    }
----------- 2016-08-18 20:24:08
why account id null [triggered by calling directly from test actions], but the same code works well in controller. 

    public static function getPeriodMissionCommonParam($mid) {
        $user = AccountManager::getCurrentUser();
        $period_mission = MissionManager::getMission($mid, MissionType::PERIOD); // exception is thrown if not existing
//        $account = Account::findIdentity(Yii::$app->user->id);


-------- 2016-08-17 13:56:48
$ du -sh * --exclude=sdb*
4.0K  android
50M AndroidStudioProjects
26M Arduino
344M  baina
592K  Desktop
4.0K  Documents
650M  Downloads
12K examples.desktop
4.0K  kidd
13M logs
4.0K  Music
20M Pictures
4.0K  Public
1.1M  ros_work
3.9G  share
23G software
4.0K  Templates
4.0K  tmp
4.0K  Videos
15G workspace
kidd@kidd-OptiPlex-9020:
~
$ df -lh
Filesystem      Size  Used Avail Use% Mounted on
/dev/sda1       110G  103G  1.8G  99% /


--------- 2016-08-17 11:55:15
/home/kidd/workspace/bc/MissionImpossible/models/PeriodMission.php
   public function getChallenges()
    {
        return $this->hasMany(Challenge::className(), ['mission_id' => 'id', 'mission_type' => MissionType::PERIOD]);
    }

public function getChallenges()
    {
        return $this->hasMany(Challenge::className(), ['mission_id' => 'id']);
//        return $this->hasMany(Challenge::className(), ['mission_id' => 'id', 'mission_type' => MissionType::PERIOD]);
    }


  public function getChallenges()
    {
        return Challenge::findAll(['mission_id' => $this->id, 'mission_type' => MissionType::SINGLE]);
//        return $this->hasMany(Challenge::className(), ['mission_id' => 'id']);
    }
----------- 2016-08-17 11:44:52

Getting unknown property: app\models\PeriodMission::checkin_min_count

 in /home/kidd/workspace/bc/MissionImpossible/models/Challenge.php at line 142 – app\models\Challenge::isSucceed()